crates/ty_python_semantic/resources/mdtest/scopes/moduletype_attrs.md

```diff
@@ -40,6 +40,37 @@ reveal_type(__dict__)
 reveal_type(__init__)
 ```
 
+## `ModuleType` globals combined with explicit assignments and declarations
+
+A `ModuleType` attribute can be overridden in the global scope with a different type, but it must be
+a type assignable to the declaration on `ModuleType` unless it is accompanied by an explicit
+redeclaration:
+
+`module.py`:
+
+```py
+__file__ = None
+__path__: list[str] = []
+__doc__: int = 42
```

Comment:
this is also unsound, I suppose? I think it would need a fix in a different area of the code, though (we just wouldn't permit the redeclaration if it's an implicit `ModuleType` attribute in the global scope and it's not assignable to the type on `ModuleType`)

Reply:
Yes, was just commenting on this. I think we could probably just forbid re-declaring these? (I also don't think this is a priority that we should spend time on now, it's not likely to impact many users. We should probably add a TODO comment for it, though.)

Reply:
(I'll add a TODO and do this as a followup if we agree)